src/arrays.js

```diff
@@ -17,22 +27,51 @@ const reduce = (elements, cb, memo) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  if (typeof memo === 'undefined') {
```

Comment:
Here you are checking for a string with the characters `'undefined'`.  You need to drop the quotes and check for the actual value `undefined`